cmd/crane/cmd/push.go

```diff
@@ -35,18 +38,31 @@ func NewCmdPush(options *[]crane.Option) *cobra.Command {
 		RunE: func(_ *cobra.Command, args []string) error {
 			path, tag := args[0], args[1]
 
-			img, err := loadImage(path)
+			img, err := loadImage(path, index)
 			if err != nil {
 				return err
 			}
 
-			return crane.Push(img, tag, *options...)
+			if i, ok := img.(v1.Image); ok {
+				return crane.Push(i, tag, *options...)
+			}
+			if idx, ok := img.(v1.ImageIndex); ok {
+				// TODO(generics): Make crane.Push support index.
+				o := crane.GetOptions(*options...)
+				ref, err := name.ParseReference(tag, o.Name...)
+				if err != nil {
+					return err
+				}
+				return remote.WriteIndex(ref, idx, o.Remote...)
+			}
+			return fmt.Errorf("cannot push type (%T) to registry", img)
 		},
 	}
+	cmd.Flags().BoolVar(&index, "index", false, "Push the collection of images as a single index")
```

Comment:
My interpretation of this text is that the fallback behavior is to push N images instead, which is not what the code does, so perhaps some clarity is called for there?